Rubberduck.Parsing/VBA/COMReferenceManager.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Rubberduck.VBEditor.SafeComWrappers.Abstract;
+using System.Collections.Concurrent;
+
+namespace Rubberduck.Parsing.VBA
+{
+    public class COMReferenceManager : COMReferenceManagerBase
+    {
+        private const int _maxReferenceLoadingConcurrency = -1;
+
+        public COMReferenceManager(RubberduckParserState state, IParserStateManager parserStateManager, string serializedDeclarationsPath = null)
+        :base(state, parserStateManager, serializedDeclarationsPath) { }
+
+
+        protected override void LoadReferences(List<IReference> referencesToLoad, ConcurrentBag<IReference> unmapped, CancellationToken token)
+        {
+            var referenceLoadingTaskScheduler = ThrottelingTaskScheduler(_maxReferenceLoadingConcurrency);
+
+            //Parallel.ForEach is not used because loading the references can contain IO-bound operations.
+            var loadTasks = new List<Task>();
+            foreach (var reference in referencesToLoad)
+            {
+                var localReference = reference;
+                loadTasks.Add(Task.Factory.StartNew(
+                                    () => LoadReference(localReference, unmapped),
+                                    token,
```

Comment:
Nearly useless. Should the cancellation happen before the task actually starts (gets its first execution slot), it will cancel right awai and not execute at all. Unfortunately, passing the token to the actiion would be useless, since the COM reflector does not (yet) support cancellation.